doc/user/content/sql/create-source/kafka.md

```diff
@@ -0,0 +1,371 @@
+`client_id`                          | `text`    | Use the supplied value as the Kafka client identifier.
+`group_id_prefix`                    | `text`    | Use the specified prefix in the consumer group ID. The resulting `group.id` looks like `<group_id_prefix>materialize-X-Y`, where `X` and `Y` are values that allow multiple concurrent Kafka consumers from the same topic.
+`ignore_source_keys`                 | `boolean` | Default: `false`. If `true`, do not perform optimizations assuming uniqueness of primary keys in schemas.
+`isolation_level`                    | `text`    | Default: `read_committed`. Controls how to read messages that were transactionally written to Kafka. Supported options are `read_committed` to read only committed messages and `read_uncommitted` to read all messages, including those that are part of an open transaction or were aborted.
+`statistics_interval_ms`             | `int`     | `librdkafka` statistics emit interval in `ms`. A value of 0 disables statistics. Statistics can be queried using the `mz_kafka_source_statistics` system table. Accepts values [0, 86400000].
+`timestamp_frequency_ms`             | `int`     | Default: `1000`. Sets the timestamping frequency in `ms`. Reflects how frequently the source advances its timestamp. This measure reflects how stale data in views will be. Lower values result in more-up-to-date views but may reduce throughput.
+`topic_metadata_refresh_interval_ms` | `int`     | Default: `300000`. Sets the frequency in `ms` at which the system checks for new partitions. Accepts values [0,3600000].
+`enable_auto_commit`                 | `boolean` | Default: `false`. Controls whether or not Materialize commits read offsets back into Kafka. This is purely for consumer progress monitoring and does not cause Materialize to resume reading from where it left off across restarts.
+`fetch_message_max_bytes` | `int` | Default: `134217728`. Controls the initial maximum number of bytes per topic+partition to request when fetching messages from the broker. If the client encounters a message larger than this value it will gradually try to increase it until the entire message can be fetched. Accepts values [1, 1000000000].
+
+## Supported formats
+
+|<div style="width:290px">Format</div> | Append-only envelope | Upsert envelope | Debezium envelope |
+---------------------------------------|:--------------------:|:---------------:|:-----------------:|
+| Avro                                 | ✓                    | ✓               | ✓                 |
+| JSON                                 | ✓                    | ✓               |                   |
+| Protobuf                             | ✓                    | ✓               |                   |
+| Text/bytes                           | ✓                    | ✓               |                   |
+|  CSV                                 | ✓                    |                 |                   |
```

Comment:
We should probably add links here to the format pages/sections when they are ready.

Reply:
That's the plan!